dali/kernels/signal/fft/fft_cpu.h

```diff
@@ -0,0 +1,138 @@
+    "Data types other than float are not yet supported");
+
+  static_assert(std::is_same<OutputType, float>::value
+             || std::is_same<OutputType, std::complex<float>>::value,
+    "Data types other than float are not yet supported");
+
+  DLL_PUBLIC virtual ~FftImpl() = default;
+
+  DLL_PUBLIC virtual KernelRequirements Setup(KernelContext &context,
+                                              const InTensorCPU<InputType, Dims> &in,
+                                              const FftArgs &args) = 0;
+
+  DLL_PUBLIC virtual void Run(KernelContext &context,
+                              const OutTensorCPU<OutputType, Dims> &out,
+                              const InTensorCPU<InputType, Dims> &in,
+                              const FftArgs &args) = 0;
+};
+
+}  // namespace impl
+
+/**
+ * @brief Computes 1-D FFT related transformation from real data to either a complex spectrum
+ *   or a transformation of the complex spectrum (power, magnitude, log power)
+ *
+ * Input data can be a 2D or 3D tensor representing an signal (e.g. [channels, time]) or
+ * a sequence of frames (e.g. [channels, frames, time]). The kernel can work with other data
+ * layouts by providing the transform_axis representing the dimension to be transformed to the
+ * frequency domain (e.g. for a layout of [channels, time, frames] we set transform_axis=1 to
+ * produce a [channels, frequency, frames] layout)
+ *
+ * @param args.spectrum_type defines the nature of the output
+ *   FFT_SPECTRUM_COMPLEX:
+ *     Output represents the complex spectrum with real and imaginary parts interleaved
+ *     Output is a 2D tensor of shape Fx(NFFT*2) where NFFT represents the FFT size
+ *   FFT_SPECTRUM_MAGNITUDE:
+ *     Output represents the magnitude of positive half of the spectrum,
+ *      as a 2D tensor of shape Fx(NFFT/2+1)
+ *   FFT_SPECTRUM_POWER:
+ *     Output represents the power of the spectrum, as a 2D tensor of shape Fx(NFFT/2+1)
+ *   FFT_SPECTRUM_LOG_POWER:
+ *     Output represents the log power spectrum, as a 2D tensor of shape Fx(NFFT/2+1)
+ * (where NFFT is the size of the FFT)
+ *
+ * @param args.nfft Number of samples in the FFT. If not provided, nfft will be calculated as the
+ *   next power of two of the size of the transform axis.
+ *
+ * @param args.transform_axis Axis along which the FFT transformation will be calculated
+ *     (note: current implementation only supports transform_axis to be the inner-most dimension
+ */
+template <typename OutputType = std::complex<float>,  typename InputType = float, int Dims = 2>
+class DLL_PUBLIC Fft1DCpu {
+ public:
+  static_assert(std::is_same<InputType, float>::value,
+    "Data types other than float are not yet supported");
+
+  static_assert(std::is_same<OutputType, float>::value
+             || std::is_same<OutputType, std::complex<float>>::value,
+    "Data types other than float are not yet supported");
+
+  DLL_PUBLIC KernelRequirements Setup(KernelContext &context,
```

Comment:
Since Setup is DDL_PUBLIC and explicitly instantiated in .cc file, it's recommended that so is the destructor - this ensures that allocator and deleter used by the impl_ unique pointer will always match.

Reply:
done